lib/fencing/st_rhcs.c

```diff
@@ -30,35 +30,27 @@ stonith__list_rhcs_agents(stonith_key_value_t **devices)
 {
     // Essentially: ls -1 @sbin_dir@/fence_*
 
-    int count = 0;
+    int count = 0, i;
     struct dirent **namelist;
-    int file_num = scandir(RH_STONITH_DIR, &namelist, 0, alphasort);
+    const int file_num = scandir(RH_STONITH_DIR, &namelist, 0, alphasort);
 
-    if (file_num > 0) {
+    for (i = 0; i < file_num; i++) {
         struct stat prop;
         char buffer[FILENAME_MAX + 1];
 
-        while (file_num--) {
-            if ('.' == namelist[file_num]->d_name[0]) {
-                free(namelist[file_num]);
-                continue;
-
-            } else if (!crm_starts_with(namelist[file_num]->d_name,
-                                        RH_STONITH_PREFIX)) {
-                free(namelist[file_num]);
-                continue;
-            }
-
-            snprintf(buffer, FILENAME_MAX, "%s/%s", RH_STONITH_DIR,
-                     namelist[file_num]->d_name);
+        if (crm_starts_with(namelist[i]->d_name, RH_STONITH_PREFIX)) {
+            /* @TODO: migrate to fstatat when POSIX 2008+ is bare minimum */
```

Comment:
We require minimum versions of library dependencies, but we don't put much requirement on the OS. We could define a wrapper in lib/common/compat.c or a replacement in replace/.

Reply:
If we don't want to wait for "time has come to impose as recent
lower portability limits", We should eventually start using Gnulib
much more extensively (and transfer and hand-made compat layers
onto that where possible).  But while definitively gaining more
versatility, security, and generally keeping with the pace of what's
considered best practices, it would be quite a massive subproject,
and in that light, special casing just a single occurrence doesn't
seem appealing (as opposed to marking eligible places for future
consideration, which I've done here).

FTR. there's indeed:
https://git.savannah.gnu.org/gitweb/?p=gnulib.git;a=blob_plain;f=modules/fstatat

Reply:
agreed